### PR TITLE
Adds course completion event and user identification by email

### PR DESCRIPTION
**Pull request by CinecaElearning**
The changes implemented here are about the following issues:
https://github.com/LearningLocker/Moodle-xAPI-Translator/issues/36
https://github.com/LearningLocker/Moodle-xAPI-Translator/issues/37

### CODE REVIEW

File: src/Events/CourseCompleted.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    /**
+     * Reads data for an event.
+     * @param [String => Mixed] $opts
+     * @return [String => Mixed]
+     * @override Event
+     */
+    public function read(array $opts) {
+        return [array_merge(parent::read($opts)[0], [
+            'recipe' => 'course_completed',
+            'course_url' => $opts['course']->url,
```
Comment (ryansmith94) [inline src/Events/CourseCompleted.php]: A lot of this can be removed by extending CourseViewed.

File: tests/EventTest.php [hunk 2]
```diff
@@ -51,6 +51,7 @@ protected function constructUser() {
             'id' => 1,
             'url' => 'http://www.example.com/user_url',
             'username' => 'Test user_name',
+            'email' => 'test@test.com',
```
Comment (ryansmith94) [inline tests/EventTest.php]: This will also need a corresponding assertion in the `assertUser` method.

File: tests/CourseCompletedTest.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    /**
+     * Sets up the tests.
+     * @override TestCase
+     */
+    public function setup() {
+        $this->event = new Event($this->repo);
+    }
+
+    protected function assertOutput($input, $output) {
+        parent::assertOutput($input, $output);
```
Comment (ryansmith94) [inline tests/CourseCompletedTest.php]: Can probably extend the CourseViewedTest to reduce code duplication.

**Comment by ryansmith94** [comment]
Just a few comments, looks good besides those comments. However, I am interested to know why so many examples were seemingly changed entirely?

**Comment by CinecaElearning** [comment]
I have to apologize, since I have to say that I really don't know. I saw them changed and I wanted to ask but, posting all this work all together, I forgot to...
As I was saying, I'm starting now using phpUnit and, when I ran the tests on this plugin, I found the files changed. I supposed they were a product of the phpUnit computation so I added them. Maybe I ran the test missing a crucial parameter? Anyway, I didn't intentionally changed them, so they can be left apart in the merging process as far as I know.

**Comment by ryansmith94** [comment]
Ok that's interesting, I don't think PHPUnit automatically updates the examples.

**Comment by CinecaElearning** [comment]
When I have a bit of time, I'll try to understand how that happened.

**Comment by jlowe64** [comment]
The PHPUnit tests now rebuild the examples automatically. It's in the EventTest file `$this->createExampleFile($output);`

When you run build.sh, it runs the unit tests, removing the old examples and adding refreshed versions. I avoid including all these new examples by keeping a dev environment where I run my tests, and don't run my tests on my repo until I am ready to do a new build. 

The testing documentation should probably be expanded a lot more to include some tips and tricks to make it a lot easier.

**Comment by ryansmith94** [comment]
Ahhhhh yes I forgot we added that, thanks @jlowe64.

**Comment by ryansmith94** [comment]
Once commit comments have been addressed this can be merged I think assuming the tests pass.

**Comment by davidpesce** [comment]
Course completion is already incorporated by other PRs. The email identifier, unfortunately, does not work. I'm creating a new issue in the main repo to address this in the future.

**Pull request closed without merging**